### ISSUE TITLE
A1 Mini Webcam

### ISSUE THREAD

**Issue by TLMcNulty** [jneilliii/OctoPrint-BambuPrinter#32]
I've done some searching but haven't found anything. The Bambu plugin and workflow works well once you understand how to use this plugin, but I'm familiar with the RTC streams in the new camera stack stuff in octoprint, and was wondering if there was any documentation on getting the A1 Mini's webcam into a platform like octoprint. May be out of scope for this plugin but I was hoping it'd be possible to add to my multicam setup.

**Comment by TLMcNulty** [comment]
A few resources I've come across:

https://github.com/synman/bambu-go2rtc
https://github.com/bambulab/BambuStudio/issues/1536#issuecomment-1811916472
https://github.com/slynn1324/BambuP1Streamer
https://github.com/synman/webcamd/tree/bambu

**Comment by jneilliii** [comment]
Yeah, synman's approach is something we kind of worked out together and probably the only way to integrate well with OctoPrint. I actually use go2rtc and point to my x1c's rtsp stream, then use the HLS link for that stream in OctoPrint's classic webcam settings.

**Comment by TLMcNulty** [comment]
Very impressive. I was able to get the stream into octoprint as I described wanting. Running the go2rtc binary gets video and opens the ports locally as described. We just need to formalize some sort of approach for defining a Bambu printer in Octoprint and setting up go2rtc.

**Comment by jneilliii** [comment]
I had it running as a service using these steps: https://gist.github.com/jneilliii/93b412cb0bbf6b7bfd76f7e10d612f24

you would just have to deal with any path differences where you saved the go2rtc binaries/config. it assumes pi username in `ExecStart`

**Comment by secco04** [comment]
@MrDrache333  hey hi thanks for that but still i have trouble to get an image out .... 
i assume the "hostname" must be the printers ip address?

**Comment by Zeunas** [comment]
> Yeah, synman's approach is something we kind of worked out together and probably the only way to integrate well with OctoPrint. I actually use go2rtc and point to my x1c's rtsp stream, then use the HLS link for that stream in OctoPrint's classic webcam settings.

Hey, any chance you could explain how to install this? https://github.com/synman/webcamd/tree/bambu
Thanks!

**Comment by kalinjul** [comment]
@MrDrache333 Your docker compose file works perfectly fine for me, thanks!
Using Printer's ip address as HOSTNAME and accesscode as PASSWORD.
